### DISCUSSION TITLE
version 24.x BETAs

### DISCUSSION THREAD

**Discussion by hvianna** [hvianna/audioMotion.js#46] [Announcements]
> Updated source code is in the `dev` branch

# version 24.3-beta.0

### Added:

+ Option to select a local folder as source of background images and videos, in [Config - General settings](docs/users-manual.md#general-settings) (requires [File System Access API support](docs/known-issues.md));
+ More new options in [Config - General settings](docs/users-manual.md#general-settings):
  + customize how many individual media files appear as selectable backgrounds;
  + remember the contents of the play queue between sessions;
  + enable/disable showing video files in the file explorer;
+ Enabled some server-side configuration options via `config.json` file - see [Self-hosting audioMotion](docs/server.md).

### Changed & improved:

+ The play queue (and stored playlists) can now mix files hosted on the server with ones from the local device;
+ Minor changes to default settings: Scale-Y off and track info displayed at both the start and end of the song;
+ Increased font sizes in the player main panel and help screen;
+ [Updated documentation](https://audiomotion.app/docs).

# version 24.2-beta.1

### Added:

+ Save and load playlists in local mode (File System API) - these are stored in the browser's IndexedDB storage;
+ **Prism (legacy)** and **Rainbow (legacy)** gradients - disabled by default, enabled them in the Config panel;
+ Basic support for video file extensions (audio playback only) - thanks **@JesusGod-Pope666-Info** for the [suggestion](https://github.com/hvianna/audioMotion.js/discussions/40).

### Improved:

+ Fixed retrieving metadata for queued songs when using the File System API;
+ Improved autoplay start time when adding an m3u playlist to the play queue;
+ Update the right-channel gradient selection when the LINK button is activated.

# version 24.2-beta.0 

A little over two years ago, I started working on turning audioMotion into a native application using **Electron**, but then I realized it would be a pain to keep up with the constant framework updates and huge dependency tree.

Then, I wanted to believe **File System Access API** was gaining traction for wider browser adoption and thought it would be really cool to have an official audioMotion web app that everyone could use to play their own music without needing to download or install anything, while also being easier for me to maintain. So I began adding support for that too. Turns out, it's 2024 and this API is still supported by Chromium only.

In the meantime, I've found out that some people were actually using audioMotion as a self-hosted web app and, since this is also how I use it myself on my old NAS, I wanted to keep the app working on any standard web server as well.

**TL;DR -** audioMotion code is now more complex than ever, but lately I simply haven't had the time or energy to work on it as much as I'd like to, so I still don't have a stable version working 100% as I want.

However, this version does have a lot of new cool features added since the last public release, and my wife and I have been using it at home, and she keeps telling me I need to publish it.

So.. there you have it! **Click the button below to open the app:**

[![Button Icon]][Link]

[Link]: https://audiomotion.app 'click to open web app'

[Button Icon]: https://img.shields.io/badge/audioMotion.app-ff3e37?style=for-the-badge&logoColor=white&logo=Soundcharts

> **⚠️ This is BETA software! Expect bugs!**

### Known limitations:

+ Access to local device works only on Chromium-based browsers, due to currently limited [File System Access API support](https://caniuse.com/native-filesystem-api)<br>should work on Chrome and Edge by default; on Brave you'll need to enable it in `chrome://flags/#file-system-access-api`
+ Saving and loading playlists on File System API mode doesn't work yet;
+ Documentation hasn't been updated yet.

### Added:

+ Support to File System Access API, allows the web app to play music from your local device;
+ *Bark*, *Mel* and linear frequency scales for improved visualization of mid-range and high frequencies;
+ Frequency weighting filters;
+ New dual-channel layouts: horizontal (side-by-side channels) and combined (overlaid channels);
+ New bar coloring modes: by level (bar amplitude) and by index (bar position);
+ Rounded corner bars (optional);
+ Option to display musical note labels in the X-axis scale;
+ Option to use linear amplitude values, instead of dB;
+ Option to use IEC/ANSI preferred frequencies for octave bands;
+ Allow selecting different gradients for the left and right channels, when using a dual-channel layout;
+ Playlists can now be saved to the hard disk (native app only).

### Changed / improved:

+ The **Prism** and **Rainbow** gradients have been updated with softer colors;
+ The custom preset funcionality now provides 9 slots for saving your favorite presets;
+ The **Smoothing** setting has been moved to [General settings](users-manual.md#general-settings), and the default value changed to 0.7 (from 0.5);
+ Random Mode has been renamed to **Randomize** and it's now possible to remove the *Mode* from the settings randomized;
+ The **AUTO** gradient button/functionality has been removed - check *Gradients* under [Settings affected by Randomize]() to have new gradient(s) selected when *Randomize* is active;
+ The **STEREO** button has been replaced by the **Channel Layout** setting;
+ Added more options to the minimum and maximum values in the frequency range;
+ New fonts for the user interface and internal console;
+ New custom UI controls to replace standard range inputs and radio buttons;
+ audioMotion can now run from a subdirectory in web server mode.

### Fixed:

+ Reflex not working on PIP mode on some circumstances.

<hr /><em>This discussion was created from the release <a href='https://github.com/hvianna/audioMotion.js/releases/tag/v24.2-beta.0'>version 24.2-beta.0</a>.</em>

**Comment by JesusGod-Pope666-Info** [comment]
Awesome is there a way yet to point out a specific folder for playing on the website and not just the usual folder?

The Setting screen jumps out to me at first sight, looks nice. Looks better organized and all, looks much more clear.

Yea I had some Gradient that I made for the player in true rainbow yet lost them as I guess the data was stored in the browser :/
Any other place I can do it? I had 2 favorites that I made with my specific 7 rainbow colors, and sadly lost them. I guess I need to go into some further coding somewhere then in one of the files where you must have those standard Gradients placed.

The new text in the music and play folder, I kinda do not like, at least not on first look - the older looks better and easier to read for me. But maybe there is something 80 with the new text, dunno.... But it does seems somewhat strange to my eyes.

-- Darkijah

**Reply by hvianna** [reply]
> Awesome is there a way yet to point out a specific folder for playing on the website and not just the usual folder?

Is the folder accessible via web (http)? If it is, you may be able to redirect accesses to the `music/` folder to the new address, adding the lines below to your `.htaccess` file:

```htaccess
RewriteEngine on
RewriteRule ^/music/(.*)$ http://example.com/somefolder/$1 [NC,L,R=301]
```

If you can't access the folder via web you'll need to create an alias for `music/`, pointing to the desired folder. But that can't be done so easily via `.htaccess`, you'll need further access to your web server configuration. This is documented [here](https://audiomotion.me/#/server?id=configuration-tips)

> The Setting screen jumps out to me at first sight, looks nice. Looks better organized and all, looks much more clear.

Glad to hear that!

> Yea I had some Gradient that I made for the player in true rainbow yet lost them as I guess the data was stored in the browser :/ Any other place I can do it? I had 2 favorites that I made with my specific 7 rainbow colors, and sadly lost them. I guess I need to go into some further coding somewhere then in one of the files where you must have those standard Gradients placed.

Yes, custom gradients are stored in the browser, but only the site that created them can "see" them. So, your gradients should still be there, but you'll need to create them again if you're using audiomotion.app

> The new text in the music and play folder, I kinda do not like, at least not on first look - the older looks better and easier to read for me. But maybe there is something 80 with the new text, dunno.... But it does seems somewhat strange to my eyes.

Thanks for the input. The old version uses the system font, but the problem is that each operating system looks different. I tried to find a good external font that would make the player look consistent on any system. Maybe I need to give it another try.

**Reply by JesusGod-Pope666-Info** [reply]
Well.... So I fixed the http://JesusGod-Pope666.Info/content/Media with an .htaccess with Options +Indexes.
![Screenshot_2024-02-28_22-26-09](https://github.com/hvianna/audioMotion.js/assets/90806094/a676ba64-0e67-43fb-9120-bef849d11aeb)
And can be seen here: http://JesusGod-Pope666.Info/content/Media

So I need to put the rewrite code in the .htaccess in the music folder? I have done so, and it does not seem to do anything but break the access:

RewriteEngine on
RewriteRule ^/music/(.*)$ http://JesusGod-Pope666.Info/content/Media/$1 [NC,L,R=301]

if I put the Options +Indexes into it I just get the usual.

Any ideas?

-- Darkijah

**Reply by hvianna** [reply]
The .htaccess with the rewrite rules must be in the same folder of audioMotion's index, not inside the music folder.

**Reply by JesusGod-Pope666-Info** [reply]
Does not seem to work as far as I can gather.

Nothing else then this in the root folder of audioMotion? No need of any other things? Does not seem to do much.
RewriteEngine on
RewriteRule ^/music/(.*)$ http://jesusgod-pope666.info/content/Media/$1 [NC,L,R=301]

**Reply by JesusGod-Pope666-Info** [reply]
Well I was looking around... I did find: `Redirect 301 /music http://jesusgod-pope666.info/content/Media` which works for redirecting from music to the specific folder in the browser, but does not work in the audioMotion player as it does not reach anything.

**Reply by JesusGod-Pope666-Info** [reply]
Not sure what I have done but the music folder is kabut..... Showing nothing now. It still seem to be directing to the other place even if I have deleted the .htaccess files or replaced them with others.

**Reply by hvianna** [reply]
> Not sure what I have done but the music folder is kabut..... Showing nothing now. It still seem to be directing to the other place even if I have deleted the .htaccess files or replaced them with others.

The redirect is cached by the browser, so you'll need to clear the cache (Ctrl+Shift+Delete) to get the new rules working (closing and reopening the browser should also work).

Try the rules below, this time **inside audioMotion's music folder:** (remove the other redirect in root folder)

```
RewriteEngine on
RewriteRule ^(.*)$ http://jesusgod-pope666.info/content/Media/$1 [NC,L,R=301]
```

**Reply by JesusGod-Pope666-Info** [reply]
ARGH it caches the thing as well, like the insanity of it all, I thought this was server sided and nothing to do with the browser.. yea great, I just deleted all the music folder and reuploading it all but it seems it just needed a total refresh. In the SeaMonkey browser you can at least turn this Cache nonsense OFF, it is such a pain having turned on when you make websites like the old days and you change something which does not change because it stores some of the data.

Okay.... Lets try again - after it reuploads the stuff, I will try it out. I do wonder if anything I did, toying around before hand actually worked or not. It like all those things tested, and the cache was most likely ruining my testing. I just did not think this had anything to do with the browser cache. Thanks for the reminder - I should have realized it myself, and yet nothing came up in the brain.

**Reply by JesusGod-Pope666-Info** [reply]
As far as I can see, it works in the browser going from http://am.jesusgod-pope666.info/music or http://am.jesusgod-pope666.info/music/ to http://jesusgod-pope666.info/content/Media/ - so far so good - but it does not seem to work in the http://am.jesusgod-pope666.info audioMotion player not being able to find any data. ( Cannot access media folder (/music) on server! )

**Reply by JesusGod-Pope666-Info** [reply]
With all the nonsense going on, I finally, at least on the mobile got audioMotion to find another folder!!!! Took a break to the Poop room with my mobile.... And surprisingly enough found it had connected with the folder in the audioMotion.
Redirect 301 /music /backgrounds
Just put some music into the backgrounds folder and find them. Well that is on the mobile.... I will now try to see if I can get it working on the computer. Naaa..... Just seems to tell me it cannot access the media folder..... Does not make any sense....

**Reply by JesusGod-Pope666-Info** [reply]
Well... this works on the mobile, on an old version of Firefox as I normally use, I have to test the other thing you said as well.... But on the Computer Browsers, nothing, the audioMotion player just tells me it can't access the media folder. But it clearly works on the mobile. I just made a test folder, music-htaccess for a beginning in the same place, just to have some of the factors gone.

Options +Indexes
Redirect 301 /music /music-htaccess

Hmmmm..... interestingly enough a more up2date browser on the mobile, seems to have the same issue. I guess more security bullshit as it usual goes for breaking things and making easy peasy things hard to accomplished.

**Reply by JesusGod-Pope666-Info** [reply]
```
# All folders availible:
Options +Indexes
# - - - -
Redirect 302 /music /music-htaccess/
#
# - - - - Redirect:
#Redirect 302 /music /music-htaccess/ *WORKING*
#Redirect 302 /music http://jesusgod-pope666.info/content/Media
#Redirect 302 /music https://jesusgod-pope666.info/content/Media
#RewriteEngine on
#RewriteRule ^(.*)$ http://jesusgod-pope666.info/content/Media/$1 [PT,NC,L,R=302]
#RewriteRule ^/music/(.*)$ /music-htaccess/$1 [NC,L,R=302]
#RewriteRule ^/music/(.*)$ http://jesusgod-pope666.info/content/Media/$1 [NC,L,R=302]
#
#Try adding a slash before mydir 
#RewriteRule ^/mydir/(.*)$ /mydir-and-more/$1 [R=302,NC,L]
#
#Thats because your regex pattern doesn't match mydir without the trailing slash. Try with RewriteRule 
# ^/mydir[/]?(.*)$ /mydir-and-more/$1 [R=302,NC,L]

#
##302 REDIRECTS
#Options +FollowSymLinks
#RewriteRule ^mydir/(.*)$ /mydir-and-more/$1 [R=302,NC,L]
#
## Redirect all to HTTPS
#RewriteEngine On
#RewriteCond %{SERVER_PORT} 80
#RewriteRule ^(.*)$ https://example.org/$1 [R]
## End redirect
#

```

**Reply by JesusGod-Pope666-Info** [reply]
After testing... Sigh.

```
# All folders availible:
Options +Indexes
# - - - -
RewriteEngine on
RewriteRule ^music/(.*)$ http://jesusgod-pope666.info/content/Media/$1 [NC,L,R=302]
#
# - - - - Redirect:
#Redirect 302 /music /music-htaccess/ *WORKING ON ANDROID OLD FIREFOX*
#Redirect 302 /music http://JesusGod-Pope666.Info/content/Media *NOT WORKING*
#
#RewriteEngine on
#RewriteRule ^music/(.*)$ /music-htaccess/$1 [NC,L,R=302] *WORKING ON ANDROID OLD FIREFOX*
#RewriteRule ^music/(.*)$ http://JesusGod-Pope666.Info/content/Media/$1 [NC,L,R=302] *NOT WORKING*
#
```
But beside the older Firefox browser, it does not work in the audioMotion player - and these are the only 2 code that worked. But.... If it worked on the other browser, I guess one could put it directly into the main page URL and then Redirect it, if it actually worked every place beside the old Firefox browser on my mobile.

Anyone have any ideas, by all means.

**Reply by hvianna** [reply]
The redirect is done in the server, it's not dependent on browser. Except for the fact that browsers cache the redirects, so this is kind of a PITA to test. Every time you change something you'll need to close the tab you're using and clear the cache or close the browser. Try accessing again, press Ctrl+F5 to force a page refresh.

[The solution I've sent before](https://github.com/hvianna/audioMotion.js/discussions/46#discussioncomment-8630016) worked in my server.

**Reply by JesusGod-Pope666-Info** [reply]
Checking the background stuff:

audioMotion.js:1 
        
        
       GET http://am.jesusgod-pope666.info/playlists.cfg 404 (Not Found)
Bt @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
8169 @ audioMotion.js:1
o @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
o.O @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
audioMotion.js:1 
        
        
       GET http://am.jesusgod-pope666.info/serverInfo 404 (Not Found)
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
8169 @ audioMotion.js:1
o @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
o.O @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
am.jesusgod-pope666.info/:1 Access to fetch at 'http://jesusgod-pope666.info/content/Media/' (redirected from 'http://am.jesusgod-pope666.info/music/') from origin 'http://am.jesusgod-pope666.info' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
audioMotion.js:1 
        
        
       GET http://jesusgod-pope666.info/content/Media/ net::ERR_FAILED
(anonymous) @ audioMotion.js:1
d @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
Promise.then (async)
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
8169 @ audioMotion.js:1
o @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
o.O @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
audioMotion.js:1 Error accessing directory. TypeError: Failed to fetch

Seems to be some security nonsense again..... Like the Radio channels being block by this bullshit known as CORS.

**Reply by JesusGod-Pope666-Info** [reply]
audioMotion.js:1 
        
        
       GET http://am.jesusgod-pope666.info/playlists.cfg 404 (Not Found)
Bt @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
8169 @ audioMotion.js:1
o @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
o.O @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
audioMotion.js:1 
        
        
       GET http://am.jesusgod-pope666.info/serverInfo 404 (Not Found)
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
8169 @ audioMotion.js:1
o @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
o.O @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
am.jesusgod-pope666.info/:1 Access to fetch at 'http://jesusgod-pope666.info/content/Media/' (redirected from 'http://am.jesusgod-pope666.info/music/') from origin 'http://am.jesusgod-pope666.info' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
audioMotion.js:1 
        
        
       GET http://jesusgod-pope666.info/content/Media/ net::ERR_FAILED
(anonymous) @ audioMotion.js:1
d @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
Promise.then (async)
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
8169 @ audioMotion.js:1
o @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
o.O @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
audioMotion.js:1 Error accessing directory. TypeError: Failed to fetch

**Reply by JesusGod-Pope666-Info** [reply]
am.jesusgod-pope666.info/:1 Access to fetch at 'http://jesusgod-pope666.info/content/Media/' (redirected from 'http://am.jesusgod-pope666.info/music/') from origin 'http://am.jesusgod-pope666.info/' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
audioMotion.js:1

am.jesusgod-pope666.info/:1 Access to fetch at 'http://jesusgod-pope666.info/content/Media/' (redirected from 'http://am.jesusgod-pope666.info/music/') from origin 'http://am.jesusgod-pope666.info/' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
audioMotion.js:1

Yea, okay... that seems to be the issue. So I need to somehow turn off CORS, whatever it is, or make it numb while using audioMotion. Trying to see if I can find something about it, beside being some kind of security depravity.

**Reply by hvianna** [reply]
Oh, CORS again. I wasn't expecting that.

But since these are both on your own server, maybe you can add the necessary configuration to allow request from any origin. See https://enable-cors.org/server_apache.html

**Reply by JesusGod-Pope666-Info** [reply]
seems fitting
![Screenshot_2024-02-29_22-39-11](https://github.com/hvianna/audioMotion.js/assets/90806094/6d90fb44-d783-4149-b0ce-33c75fe78dbd)

**Reply by JesusGod-Pope666-Info** [reply]
So I need to put response.addHeader("Access-Control-Allow-Origin", "*"); somewhere, I guess.....
I think I have toyed around with this before.... but I don't recall anything, maybe it was when I was doing something with iframe.

**Reply by JesusGod-Pope666-Info** [reply]
`<?php header("Access-Control-Allow-Origin: *"); ?>`
![Screenshot_2024-02-29_22-53-25](https://github.com/hvianna/audioMotion.js/assets/90806094/bab1328a-52df-44c8-a9ca-96c88e97548e)
https://www.aurigma.com/docs/us8/enabling-cross-origin-resource-sharing.htm#configuringHeader

**Reply by JesusGod-Pope666-Info** [reply]
FINALLY!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Well.... At least it looks like something worked. *Sigh*
![Screenshot_2024-02-29_23-26-09](https://github.com/hvianna/audioMotion.js/assets/90806094/abbf2bd9-5ccb-4e8f-b14e-141c098aed3a)

**Reply by JesusGod-Pope666-Info** [reply]
Seems like it works now... Some CORS nonsense.... Well, whatever.... Although it does not show the mp4 files, but you have fixed that in the newer version as I understand - I guess, I can put a 0 on the main files and have them at top and then the rest as is.... Whatever the case, a good amount of folders and files are now are seen that is connected to the other player. That is pretty cool.

**Reply by JesusGod-Pope666-Info** [reply]
Beside not finding all the mp4 files.... It seems to work, check it out here: http://am.jesusgod-pope666.info/ - The main files that was in the audoMotion player has been giving a - to keep them at the top, and then the rest of the folders from the other comes around. Seems to work as far as far as I can see.

**Reply by JesusGod-Pope666-Info** [reply]
Have been sorting out stuff, lots of hours of sorting as usual - never gets done it seems. But overall, it is better then it was - so moving forward. I do wonder if anyone is actually using what I make, or if it is a waste of time. Hopefully someone gets some usage out of it! A good day of work, I guess, not sure if there is any issues with the linking to the other folder, but at least I can test things out without being blocked. That is at the very least, one wall jump for freedom.

So when you new version is ready, I guess all the MP4 files will then be ready to play - although I really wonder if you could not make it play MP4 files as well. You should have a lot of puzzle pieces that could get it to work. That would be a pretty heavy free alternative for a all around Media Player. Although the beginning package could be better for beginners for sure - had a lot of starting difficulty which could have been avoided if it just had the standard stuff to get started quickly.

Anyway, it is free.... But it does make more use it, if it is easy to begin with and start of :)

I guess it is time for bed, morning here in Denmark, Grey sky as usual outside - 250 forms of grey..... Anyway, I will get back to you and others on how I got the folder link working. Although I guess it makes sense to some people, yet took me hours and hours and hours to sort it out. Although I need to test if there is some issues, I have done F5 a couple of times, not sure if that is an overall issue or what. I should try some of the other ideas and see if that works better if so. Whatever the case, it works for the most part, pretty good.

- Darkijah

**Reply by JesusGod-Pope666-Info** [reply]
[hvianna](https://github.com/hvianna)
I fixed your 2 broken Radio stations - they have CORS issues on my end. So sorted it out if that is the stations you want. *Hack the planet as it goes* Although who knows how long they work, they tend to lock here and there with there security nonsense or country walls.
There is a good amount of worldly Radio stations that I have posted as well on this forum - there is also some 80's channels if you want some kind of sound to match your players style.

[Jazz and Dance Wave Radio fixed.zip](https://github.com/hvianna/audioMotion.js/files/14457399/Jazz.and.Dance.Wave.Radio.fixed.zip)

Actually... I wanted to test if they worked in https as I forgot and hit into issues. Not only with one of the Radio stations but with the folder referral.... Yea well issues for another day. Apparently Dance Wave is stubborn in https, tried to hack the planet here and there with no success. But it does work in http mode.

**Comment by JesusGod-Pope666-Info** [comment]
seems your website forces you to httpS and don't let you do http - might be an issue for a good amount of radio channels sadly.
httpS makes a lots of further issues as it usually is with all the security and walling nonsense on the internet.

Has some issues adding my Christian playlist, although it adds the non Christian without issues. Anyway it is a Beta, I understand.

**Comment by JesusGod-Pope666-Info** [comment]
@hvianna Any idea? Backgrounds don't load in httpS mode - I don't recall having this issue before, but hard to recall.... whatever the case, it at the very least now don't load the backgrounds with httpS, and I can't remember if it was an issue before I was toying around with htaccess stuff which I am still at.


s @ vendors.js:2
(anonymous) @ audioMotion.js:1
8169 @ audioMotion.js:1
o @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
o.O @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
(anonymous) @ audioMotion.js:1
am.jesusgod-pope666.info/:593 Mixed Content: The page at 'https://am.jesusgod-pope666.info/' was loaded over HTTPS, but requested an insecure resource 'http://am.jesusgod-pope666.info/backgrounds/'. This request has been blocked; the content must be served over HTTPS.

**Comment by JesusGod-Pope666-Info** [comment]
Done a lot with the files to fit them to the player - still lot of work, but surely closer to something useful now.

Looking forward to your stable version and mp4 support! :)

**Comment by JesusGod-Pope666-Info** [comment]
@hvianna Could you do, so the user decides how many backgrounds there is available? Like if I want like 50 small backgrounds or something it would be nice if the user then had the choice to do so, or even more! :) I just love having the user have the ability to have the choice as much as possible and have the control.

Would there be a problem with making the app use more backgrounds? I don't see a big problem with it, but I am not a programmer. But would be nice for the user to decide how many backgrounds he wants in it.

Thanks, and how is it coming up with the newest version, are we, well you getting close to a stable version?

Thanks maestro! :) 
-- Darkijah

**Reply by hvianna** [reply]
Yea, I can do that. Options are good, I like them too :)  Sometimes I'm afraid audioMotion already has way too many options, it gets confusing for the user. But not in this case, I think this is good. The only issue I can see is a huge amount (like 100+) of backgrounds resulting in an impractically long selection box. But let's let the user decide! :)

The new version is coming along, made some progress this week. I'll probably release an updated demo this weekend (will try to squeeze the backgrounds option in) and am aiming for next month to release the stable.

**Reply by JesusGod-Pope666-Info** [reply]
Awesome!

**Comment by JesusGod-Pope666-Info** [comment]
@hvianna Another thing, as written other place I think - you might already have the possibility to actually also make it work as video player as well. That could be pretty cool as well. You could maybe fix something smart, or make kinda, a minor thing, like a small window running where the picture is if it is mp4 video, and maybe make the possibility for the user to make it fullscreen if they want. Now that probably takes some programming and skills - although I don't think you are lagging in that as far as I can see.... the question is usually time.

Just a suggestion, could be pretty cool feature overall.

**Reply by hvianna** [reply]
Yup, already on my to-do list. It's just not that easy. I'll have to put some more thinking on that, but it's noted.

**Reply by JesusGod-Pope666-Info** [reply]
And another Awesome! :) I am working on getting things sorted out for the player of Material, little by little, background pictures and such. Will be interesting to see things with the MP4, but there is a lot of work to make file name smaller and such.

It seems the link from the Music folder to the content Media folder works wonderful - just a little wait some times, but it seems to work and not break which I was afraid it would do. But seems to work without much issues.
